hilti/toolchain/include/ast/expressions/keyword.h

```diff
@@ -19,14 +19,15 @@ enum class Kind {
     Self,         /**< `self` */
     DollarDollar, /**< `$$` */
     Captures,     /**< `$@` */
-    Scope         /**< `$scope`*/
+    Scope,        /**< `$scope`*/
+    Stream        /**< `$stream`*/
```

Comment:
I do understand where you want to take `$stream` yet. It would seem to me that it should be possible to implement this as a free-standing accessor function in `spicy`, e.g., `spicy::stream` (we currently have e.g., `spicy::current_time` as a similar construct). Do you have more functionality around `$ID`-style identifiers in mind or is this inspired by [EVT-style variables](https://docs.zeek.org/en/master/devel/spicy/reference.html#:~:text=likewise%20left%20unset.-,In%20addition%20to%20full,only%20with%20protocol%20analyzers.,-Note) -- I believe there we had a syntax need for them to make clear they were neither Spicy nor Zeek).

Reply:
Unless we want to provide direct access to the stream in the future we could probably get by with e.g., `spicy::stream_statistics()`.

Reply:
> I picked `$stream`

An in-between syntax would be making it a unit method: `self.stream()`.  That would make it more like `offset()`. How about that?

Reply:
> An in-between syntax would be making it a unit method: self.stream(). That would make it more like offset(). How about that?

I'd like that much more than a magic `$stream` since it would make it clear that this only returns a useful result in a unit's context (also: less magic syntax to pick up 😉).